hkmc2/shared/src/main/scala/hkmc2/semantics/Symbol.scala

```diff
@@ -155,6 +157,17 @@ class ClassSymbol(val tree: Tree.TypeDef, val id: Tree.Ident)
   def nme = id.name
   def toLoc: Option[Loc] = id.toLoc // TODO track source tree of classe here
   override def toString: Str = s"class:$nme"
+  /** Compute the arity. */
+  def arity: Int = defn match
+    case S(d) => d.paramsOpt.fold(0)(_.length)
+    case N => tree.paramLists.headOption.fold(0)(_.fields.length)
+  /** Get parameter symbols. Create new symbols if not elaborated. */
+  override lazy val params: Ls[LocalSymbol & NamedSymbol] = defn match
+    case S(d) => d.paramsOpt.fold(Nil)(_.map(_._2))
+    case N => tree.paramLists.headOption.fold(Nil): tup =>
+      tup.fields.iterator.flatMap(_.param).map:
+        case (id, _) => TermSymbol(ParamBind, S(this), id)
+      .toList
```

Comment:
What is the rationale for this complicated logic? If someone somehow calls this while the elaborator is in the process of computing the `defn`, we'll end up with duplicated symbols. It's just a pretty bad/error-prone design.

Why not follow what I suggested? The symbols should be created in `TypeDefImpl`.